src/Shape/Box.js

```diff
@@ -0,0 +1,385 @@
+      @chainable
+  */
+  render() {
+
+    if (this._select === void 0) {
+      this.select(select("body").append("svg")
+        .style("width", `${window.innerWidth}px`)
+        .style("height", `${window.innerHeight}px`)
+        .style("display", "block").node());
+    }
+
+    const outlierData = [];
+    let orient = "vertical";
+
+    const filteredData = nest()
+      .key((d, i) => this._orient(d, i) === "vertical" ? this._x(d, i) : this._y(d, i))
+      .entries(this._data)
+      .map((d, i) => {
+        d.data = merge(d.values);
+        orient = this._orient(d.data, d.i);
+        const values = orient === "vertical" ? d.values.map(this._y) : d.values.map(this._x);
+
+        values.sort((a, b) => a - b);
+        d.i = this._data.indexOf(d.values[0]);
+
+        d.first = quantile(values, 0.25);
+        d.median = quantile(values, 0.50);
+        d.third = quantile(values, 0.75);
+
+        // Compute values for vertical orientation.
+        if (orient === "vertical") {
+          d.height = d.third - d.first;
+          d.width = this._rectWidth(d.data, d.i);
+          d.x = this._x(d.data, d.i);
+          d.y = (d.third + d.first) / 2;
+          d.medianY = d.y - d.height / 2 + d.third - d.median;
+        }
+        else if (orient === "horizontal") {
+        // Compute values for horizontal orientation.
+          d.height = this._rectWidth(d.data, d.i);
+          d.width = d.third - d.first;
+          d.x = (d.third + d.first) / 2;
+          d.y = this._y(d.data, d.i);
+          d.medianX = d.x + d.width / 2 - d.third + d.median;
+        }
+
+        const mode = this._whiskerMode;
+
+        if (mode[0] === "tukey") {
+          d.top = d.third + (d.third - d.first) * 1.5;
+          if (d.top > max(values)) d.top = max(values);
+        }
+        else if (mode[0] === "extent") d.top = max(values);
+        else if (typeof mode[0] === "number") d.top = min([max(values), quantile(values, mode[0] / 100)]);
+
+        if (mode[1] === "tukey") {
+          d.bottom = d.first - (d.third - d.first) * 1.5;
+          if (d.bottom < min(values)) d.bottom = min(values);
+        }
+        else if (mode[1] === "extent") d.bottom = min(values);
+        else if (typeof mode[1] === "number") d.bottom = max([min(values), quantile(values, mode[1] / 100)]);
+
+        // Compute data for outliers.
+        d.values.forEach((eachValue, index) => {
+          const value = orient === "vertical" ? this._y(eachValue, index) : this._x(eachValue, index);
+         
+          if (value < d.bottom || value > d.top) {
+            const dataObj = {};
+            dataObj.__d3plus__ = true;
+            dataObj.data = eachValue;
+            dataObj.i = index;
+            dataObj.outlier = this._outlier(d, i);
+
+            if (orient === "vertical") {
+              dataObj.x = d.x;
+              if (value < d.bottom) {
+                dataObj.y = d.y + d.height / 2 + d.first - d.bottom + value;
```

Comment:
If the data point is an outlier, aren't the `x` and `y` positions directly calculated from the `this._x` and `this._y` accessors? Meaning all of these calculations for `y` if vertical and `x` if horizontal are necessary.